crates/kubelet/src/volume/mod.rs

```diff
@@ -88,8 +109,12 @@ impl Deref for Ref {
 
 impl Drop for Ref {
     fn drop(&mut self) {
-        if matches!(self.volume_type, Type::ConfigMap | Type::Secret) {
-            // TODO: Currently there is no way to do this async (though there is an async destructors proposal)
+        if matches!(
+            self.volume_type,
+            VolumeType::ConfigMap | VolumeType::Secret | VolumeType::PersistentVolumeClaim
```

Comment:
I missed this the first time. I actually don't think we can let this clean up the persistent volume because otherwise it will be removing the data from the PV. The PV needs to be unmounted before we clean up the directory. I am guessing the volume mounter takes care of this cleanup and even if it doesn't, we can make it do it later when we try to port it to Rust. For now I am fine leaving an orphaned directory if that is the case

Reply:
Fair enough. And I think leaving an orphaned directory is fine since it's a `mount`/`umount` anyways